base/deprecated.jl

```diff
@@ -191,4 +191,13 @@ MPFR.BigFloat(x::Real, prec::Int, rounding::RoundingMode) = BigFloat(x, rounding
     Base.@deprecate_binding Mutex ReentrantLock
 end
 
+
+@eval Math begin
+    macro horner(x, p...)
+        @warn "The `@horner` macro is deprecated. Use the `evalpoly` function instead."
```

Comment:
If we deprecate this, probably code that uses `@horner` should use `@evalpoly` to stay compatible with all 1.x releases?

Reply:
Right, that's a good point.

Reply:
Neither does `evalpoly`, I thought?

Hopefully all of the uses of this macro are for real arguments (or don't care if a faster method is used for complex).

Reply:
In the above macro, I invoke the method `evalpoly(::Any, ::Tuple)` to bypass dispatch and just use horner’s method. But yeah, in practice it may not matter to anyone if the macro actually uses Horner or not.